src/lib/components/VideoCard/VideoCard.svelte

```diff
@@ -17,17 +34,43 @@
       console.warn("Unsupported video type; could not embed video.");
     }
   }
+
+  afterUpdate(async () => {
+    if (youtubeVideoId && (!title || !description)) {
+      const snippet = await getYoutubeVideoData(youtubeVideoId);
+      title = snippet.title;
+      description = snippet.description;
```

Comment:
Small nit that doesn't need to block the PR: I think it would be cleaner to make these separate variables instead of having the component reassign its own props.